src/CSVerifierBiFork.sol

```diff
@@ -0,0 +1,385 @@
+    GIndex public immutable GI_FIRST_VALIDATOR_PREV;
+
+    /// @dev This index is relative to a state like: `BeaconState.validators[0]`.
+    GIndex public immutable GI_FIRST_VALIDATOR_CURR;
+
+    /// @dev This index is relative to a state like: `BeaconState.historical_summaries`.
+    GIndex public immutable GI_HISTORICAL_SUMMARIES;
+
+    /// @dev The very first slot the verifier is supposed to accept proofs for.
+    Slot public immutable FIRST_SUPPORTED_SLOT;
+
+    /// @dev The first slot of the currently compatible fork.
+    Slot public immutable PIVOT_SLOT;
+
+    /// @dev Lido Locator contract
+    ILidoLocator public immutable LOCATOR;
+
+    /// @dev Staking module contract
+    ICSModule public immutable MODULE;
+
+    error RootNotFound();
+    error InvalidGIndex();
+    error InvalidBlockHeader();
+    error InvalidChainConfig();
+    error PartialWitdrawal();
+    error ValidatorNotWithdrawn();
+    error InvalidWithdrawalAddress();
+    error UnsupportedSlot(Slot slot);
+    error ZeroLocatorAddress();
+    error ZeroModuleAddress();
+    error InvalidPivotSlot();
+
+    /// @dev The previous and current forks can be essentially the same.
+    constructor(
+        address locator,
+        address module,
+        uint64 slotsPerEpoch,
+        GIndex gIFirstWithdrawalPrev,
+        GIndex gIFirstWithdrawalCurr,
+        GIndex gIFirstValidatorPrev,
+        GIndex gIFirstValidatorCurr,
+        GIndex gIHistoricalSummaries,
+        Slot firstSupportedSlot,
+        Slot pivotSlot
+    ) {
+        if (slotsPerEpoch == 0) revert InvalidChainConfig();
+        if (module == address(0)) revert ZeroModuleAddress();
+        if (locator == address(0)) revert ZeroLocatorAddress();
+        if (firstSupportedSlot > pivotSlot) {
+            revert InvalidPivotSlot();
+        }
+
+        MODULE = ICSModule(module);
+        LOCATOR = ILidoLocator(locator);
+
+        SLOTS_PER_EPOCH = slotsPerEpoch;
+
+        GI_FIRST_WITHDRAWAL_PREV = gIFirstWithdrawalPrev;
+        GI_FIRST_WITHDRAWAL_CURR = gIFirstWithdrawalCurr;
+
+        GI_FIRST_VALIDATOR_PREV = gIFirstValidatorPrev;
+        GI_FIRST_VALIDATOR_CURR = gIFirstValidatorCurr;
+
+        GI_HISTORICAL_SUMMARIES = gIHistoricalSummaries;
+
+        FIRST_SUPPORTED_SLOT = firstSupportedSlot;
+        PIVOT_SLOT = pivotSlot;
+    }
+
+    /// @notice Verify slashing proof and report slashing to the module for valid proofs
+    /// @param beaconBlock Beacon block header
+    /// @param witness Slashing witness
+    /// @param nodeOperatorId ID of the Node Operator
+    /// @param keyIndex Index of the validator key in the Node Operator's key storage
+    function processSlashingProof(
+        ProvableBeaconBlockHeader calldata beaconBlock,
+        SlashingWitness calldata witness,
+        uint256 nodeOperatorId,
+        uint256 keyIndex
+    ) external {
+        if (beaconBlock.header.slot < PIVOT_SLOT) {
```

Comment:
we need to have a possibility to prove before `PIVOT_SLOT` as well, innit?
that's why you changed `_getValidatorGI`